Assets/_Laz/Scripts/Environment/Puzzles/Connection/Node.cs

```diff
@@ -0,0 +1,52 @@
+using System;
+
+namespace Laz
+{
+    public class Node
+    {
+        private bool _canActivate = false;
+        private bool _isActive = false;
+
+        public event Action<bool> OnCanActivateChanged;
+        public event Action<bool> OnIsActiveChanged;
+
+        public event Action OnNodeCompleted;
+
+        public bool CanActivate
+        {
+            get => _canActivate;
+
+            set
+            {
+                _canActivate = value;
+                if (OnCanActivateChanged != null)
+                {
+                    OnCanActivateChanged(value);
```

Comment:
Do you not have to call OnCanActivateChanged(value).Invoke();

Reply:
yeah i can do that.